src/hugr/validate.rs

```diff
@@ -417,22 +419,24 @@ impl<'a> ValidationContext<'a> {
 
         match from_optype.port_kind(from_offset).unwrap() {
             // Inter-graph constant wires do not have restrictions
-            EdgeKind::Const(typ) => match from_optype {
-                OpType::Module(ModuleOp::Const(val)) => {
+            EdgeKind::Const(typ) => {
+                if let OpType::Const(ops::Const(val)) = from_optype {
                     return typecheck_const(&typ, val).map_err(ValidationError::from);
+                } else {
+                    // If const edges aren't coming from const nodes, they're graph
+                    // edges coming from Declare or Def
+                    return if [OpTag::Def, OpTag::Function].contains(&from_optype.tag()) {
```

Comment:
`OpTag::Function` contains both declares and definitions
```suggestion
                    return if OpTag::Function.contains(from_optype.tag()) {
```